src/libcore/slice/mod.rs

```diff
@@ -1362,6 +1362,21 @@ impl<T> [T] {
     /// let r = s.binary_search(&1);
     /// assert!(match r { Ok(1..=4) => true, _ => false, });
     /// ```
+    ///
+    /// Inserts a new element into already sorted array while maintaining sorted
+    /// order. Then asserts that the item was inserted correctly.
+    ///
+    /// ```
+    /// let mut s = vec![0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55];
+    /// let num = 14;
+    ///
+    /// let idx = s.binary_search(&num);
+    /// if let Ok(idx) | Err(idx) {
```

Comment:
I believe this is invalid syntax and should be `if let Ok(idx) | Err(idx) = idx {`

Reply:
See error:

```
error: expected one of `=` or `|`, found `{`
 --> slice/mod.rs:1361:27
  |
8 | if let Ok(idx) | Err(idx) {
  |                           ^ expected one of `=` or `|` here
```